src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Databend Cloud
+//
+// Licensed under the Elastic License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.elastic.co/licensing/elastic-license
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use common_catalog::table::Table;
+use common_exception::Result;
+use common_storages_fuse::FuseTable;
+use futures_util::TryStreamExt;
+use opendal::EntryMode;
+use opendal::Metakey;
+use tracing::info;
+
+#[async_backtrace::framed]
+async fn do_vacuum_drop_table(
+    table: Arc<dyn Table>,
+    dry_run_limit: Option<usize>,
+) -> Result<Option<Vec<(String, String)>>> {
+    // only operate fuse table
+    if table.engine() != "FUSE" {
+        info!(
+            "ignore table {} not of FUSE engine, table engine {}",
+            table.get_table_info().name,
+            table.engine()
+        );
+        return Ok(None);
+    }
+    // storage_params is_some means it is an external table, ignore
+    if table.get_table_info().meta.storage_params.is_some() {
+        info!("ignore external table {}", table.get_table_info().name);
+        return Ok(None);
+    }
+    let fuse_table = FuseTable::try_from_table(table.as_ref())?;
+
+    let operator = fuse_table.get_operator_ref();
+
+    let dir = format!("{}/", fuse_table.meta_location_generator().prefix(),);
```

Comment:
If the prefix `db_id/table_id` not match the `fuse_table.meta_location_generator().prefix()`, we can not remove them.
It may be a table create from another snapshot_location like:
```
create table t2 (a int) SNAPSHOT_LOCATION='3746002/3747385/_ss/3311d02b049a4dfab3c0ae8defadd301_v4.mpk';
```